### ISSUE TITLE
How to generate traffic from the simulator through Free5GC?

### ISSUE THREAD

**Issue by vggonzalez** [hhorai/gnbsim#24]
Good afternoon!

I'll sum it up as much as possible. I've created a simple scenario with 2 VMs connected to each other through an internal network.
In the first VM (internal network ip - 192.168.1.1) I have Free5GC v3.0.4 properly installed, and in the 2nd VM (192.168.1.2) I have installed gnbsim.

I'm a little bit lost so I'll explain what I did step by step:
1. Running Free5GC and then registering a subscriber through the web console (default IMSI)
2. Change the addresses on ngapIPList and gtpu as indicated on your project. I've set the 2 same IPs that you say. I understand it is okay that way, is that right?
3. In the project you say to edit the configuration file and the parameters imeisv, msin, GTPuAddr, GTPuIFname and UE.url:
-Which configuration file are you talking about? Is it on gnbsim or in free5gc? I've been looking and I can't seem to find it
-In my case GTPuAddr and GTPuIFname would be, respectively, 192.168.1.2 and enp0s8. Would there be any problems with this being an internal network? I don't think so, but just in case.
-What should be the value of UE.url? I don't really understand which use I'll have to give to that url and which url it must be
4. Once all this is done, I understand that the environment has been set up. At this point, if I disconnect my VM with gnbsim from the NAT (therefore having no internet connection), what would I have to do so the core registers the VM as an UE and therefore I can access Internet from the VM through 5GC?

Waiting for your response,
Regards,
Víctor

**Comment by hhorai** [comment]
Hello,

Thank you for your interest in gnbsim.
1. The IMSI registered on free5gc web console should be matched with `(mcc)(mnc)(msin)` in `example.json`.
2. Yes, I think so. But I have never tried. I use different IP address. Please see also [wiki](https://github.com/hhorai/gnbsim/wiki).
3. 
    - It is `example.json` for your gnbsim. 
    - The IP address can be a local address. 192.168.1.2 should be OK.
    - UE.url is represented `url` field in `UE` field in `example.json`. If gnbsim is registered and established a PDU session, then gnbsim automatically start to access to the specified `url`. Please see the wiki.

4. There is nothing to do. gnbsim start to periodical access to `url` automatically. gnbsim doesn't need a capability for accessing the Internet. 

Best,

**Comment by vggonzalez** [comment]
Thank you for your response,

The answer to the 4th question is not quite clear for me.
So GnbSim is programmed to periodically access the URL given, but this URL could be, for example, www.google.com, is that correct? Therefore the simulator is accessing the Internet even if it doesn't generate any additional traffic besides the access to the URL.
Would it be possible to modify the code in a way that, instead of just accessing the URL, it generates specific kind of traffic towards that URL? For example, start downloading a video from a video server.

Regards

**Comment by hhorai** [comment]
> but this URL could be, for example, www.google.com, is that correct? 

Yes, it is. UE.url could be a global address (or FQDN) while your free5gc needs to be possible to access the Internet.

> Would it be possible to modify the code in a way that...

Yes, I think so. Please refer the issue #19 . The issue opener has achieved to access to the Internet from an external PC.
I would like to merge his code in the near future.
